packages/finance/src/coin/mod.rs

```diff
@@ -77,16 +80,21 @@ where
     }
 
     #[track_caller]
-    pub(super) fn into_coprime_with<OtherC>(self, other: Coin<OtherC>) -> (Self, Coin<OtherC>)
+    pub(super) const fn into_coprime_with<OtherC>(self, other: Coin<OtherC>) -> (Self, Coin<OtherC>)
     where
         OtherC: Currency,
     {
         debug_assert!(!self.is_zero() && !other.is_zero());
-        let gcd = self.amount.gcd(other.amount);
+        let gcd: Amount = gcd::binary_u128(self.amount, other.amount);
         debug_assert!(gcd > 0);
 
-        debug_assert_eq!(self.amount % gcd, 0);
-        debug_assert_eq!(other.amount % gcd, 0);
+        #[cfg(debug_assertions)]
+        if self.amount % gcd != 0 {
+            panic!("LHS-value's amount is not divisible by the GCD!");
+        } else if other.amount % gcd != 0 {
+            panic!("RHS-value's amount is not divisible by the GCD!");
+        }
+
```

Comment:
a simple `debug_assert!()`  would suffice!

Reply:
With all my respect, it does not. I am adding `const` modifier which doesn't play nice with `{|debug_}assert{|_eq|_ne}!` macros. If I didn't have to change it, I wouldn't have.

Reply:
a local compilation passes with not errors or warnings